client/virtual_channel.go

```diff
@@ -0,0 +1,293 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"perun.network/go-perun/channel"
+	"perun.network/go-perun/log"
+	"perun.network/go-perun/wallet"
+	"perun.network/go-perun/wire"
+)
+
+func (c *Client) fundVirtualChannel(ctx context.Context, virtual *Channel, prop *VirtualChannelProposal) (err error) {
+	parentID := prop.Parents[virtual.Idx()]
+	parent, ok := c.channels.Get(parentID)
+	if !ok {
+		return errors.New("referenced parent channel not found")
+	}
+
+	indexMap := prop.IndexMaps[virtual.Idx()]
+	err = parent.proposeVirtualChannelFunding(ctx, virtual, indexMap)
+	if err != nil {
+		return errors.WithMessage(err, "proposing channel funding")
+	}
```

Comment:
I think readability suffers when pulling the call into the if. Named variables also help with readability, I feel, because when you have multiple return values, you do not need state all of them at every return statement.

Reply:
after chat discussion, avoiding named returns as per convention